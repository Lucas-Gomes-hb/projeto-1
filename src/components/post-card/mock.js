export const postCardPropsMock = {
  title: "Alguma",
  body: "Coisa",
  id: 1,
  cover: "img/img.png",
};
