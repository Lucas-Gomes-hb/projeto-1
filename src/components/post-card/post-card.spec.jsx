import { PostCard } from ".";
import { postCardPropsMock } from "./mock";
const { render, screen } = require("@testing-library/react");

const props = postCardPropsMock;

describe("<PostCard />", () => {
  it("should render PostCard correctly", () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole("img", { name: props.title })).toHaveAttribute(
      "src",
      props.cover
    );
    expect(
      screen.getByRole("heading", { name: props.title })
    ).toBeInTheDocument();
    expect(
      screen.getByText(props.body, { name: props.title })
    ).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(<PostCard {...props} />);

    // expect(container.firstChild).toMatchSnapshot();
  });
});
