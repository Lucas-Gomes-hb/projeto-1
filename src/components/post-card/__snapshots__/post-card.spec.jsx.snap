// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Alguma"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      Alguma
    </h2>
    <p>
      Coisa
    </p>
  </div>
</div>
`;
